### ISSUE TITLE
[Linux] SpotX-Bash currently broken with v1.2.48.405

### ISSUE THREAD

**Issue by fegauthier** [SpotX-Official/SpotX-Bash#64]
### 🧐 Have you fully read the README and viewed similar issues?

- [X] I have read the README
- [X] I have read the FAQ
- [X] I have checked previous issues, open and closed
- [X] I have completed all troubleshooting steps and the issue is not present on stock but returns after re-applying SpotX-Bash

### Сountry of your account

Canada

### What is your Spotify Plan?

Free

### Do you have Spicetify installed?

No

### ℹ Computer information

```markdown
- Spotify version: 1.2.48.405
- OS: ZorinOS 17.2
- Shell: ZSH
```

### 📝 Description

1. Since this morning, Spotify doesn't work with SpotX. It works without

![image](https://github.com/user-attachments/assets/6e0e0d2d-0807-4b46-8867-f852d72baff1)

I uninstall SpotX, and Spotify works when uninstalled.

### 🖥️ Terminal Input/Output

bash <(curl -sSL https://spotx-official.github.io/run.sh)                

████╗███╗  ███╗ █████╗█╗  █╗  ███╗  ██╗ ████╗█╗ █╗
█╔══╝█╔═█╗█╔══█╗╚═█╔═╝╚█╗█╔╝  █╔═█╗█╔═█╗█╔══╝█║ █║
████╗███╔╝█║  █║  █║   ╚█╔╝██╗███╔╝████║████╗████║
╚══█║█╔═╝ █║  █║  █║   █╔█╗╚═╝█╔═█╗█╔═█║╚══█║█╔═█║
████║█║   ╚███╔╝  █║  █╔╝ █╗  ███╔╝█║ █║████║█║ █║
╚═══╝╚╝    ╚══╝   ╚╝  ╚╝  ╚╝  ╚══╝ ╚╝ ╚╝╚═══╝╚╝ ╚╝

Searching for client directory...

Found client Directory: /usr/share/spotify

Latest supported version: 1.2.48.405
Detected client version: 1.2.48.405

✔ Created backup
✔ Applied free-tier plan patches
✔ Enabled experimental features
✔ Finished

**Comment by jetfir3** [comment]
SpotX-Bash currently only supports 1.2.48.x on macOS. 
The Linux release of v1.2.48.x which came in last ~24hrs is not yet ready for SXB. 

SXB will be updated later today for Spotify v1.2.49.x on macOS and then eventually corrected for v1.2.48.x on Linux.

**Comment by jetfir3** [comment]
> 1. Since this morning, Spotify doesn't work with SpotX. It works without
> ...
> I uninstall SpotX, and Spotify works when uninstalled.

I finally got around to checking today's 1.2.48.405 update on Linux and I can't replicate the issue you're seeing. 

I'm wondering if it's a conflict with some previous app data?

Which version of Spotify was previously installed before upgrading to 1.2.48.405.. and was SpotX-Bash used on the previous version?

If you run the SpotX-Bash install and add the `-c` flag to clear cache.. any changes? 
If `-c` doesn't resolve it -- have you tried the FULL uninstall steps mentioned in the Troubleshooting guide.. either the manual or automated script? 

I'm curious what happens for you on a complete uninstall and fresh SpotX-Bash install -- ideally not launching STOCK Spotify prior to patching the client with SpotX-Bash.

**Comment by fegauthier** [comment]
Even with

`bash <(curl -sSL https://spotx-official.github.io/run.sh) -c`

I had the same results.

I have the same results with uninstallify and reinstalling using the command below

`bash <(curl -sSL https://spotx-official.github.io/run.sh) -c                                          

████╗███╗  ███╗ █████╗█╗  █╗  ███╗  ██╗ ████╗█╗ █╗
█╔══╝█╔═█╗█╔══█╗╚═█╔═╝╚█╗█╔╝  █╔═█╗█╔═█╗█╔══╝█║ █║
████╗███╔╝█║  █║  █║   ╚█╔╝██╗███╔╝████║████╗████║
╚══█║█╔═╝ █║  █║  █║   █╔█╗╚═╝█╔═█╗█╔═█║╚══█║█╔═█║
████║█║   ╚███╔╝  █║  █╔╝ █╗  ███╔╝█║ █║████║█║ █║
╚═══╝╚╝    ╚══╝   ╚╝  ╚╝  ╚╝  ╚══╝ ╚╝ ╚╝╚═══╝╚╝ ╚╝

Searching for client directory...


Warning: Client not found. Starting interactive mode...

Latest supported version: 1.2.49.439
Requested client version: 1.2.48.405

✔ Started interactive mode [enter y/n]

Download & install client 1.2.48.405 deb pkg? y
Clear client app cache? y
Enable developer mode? y
Hide non-music categories on home screen? y
Set lyrics background color to black? y

############################################################################################################################################################################## 100.0%
✔ Downloaded and installing Spotify
✔ Installed in '/usr/share/spotify'
✔ Cleared app cache
✔ Created backup
✔ Applied free-tier plan patches
✔ Enabled developer mode
✔ Enabled experimental features
✔ Removed non-music categories on home screen
✔ Finished
`

![image](https://github.com/user-attachments/assets/02d69748-b26d-415b-b0e2-85747596321c)

**Comment by jetfir3** [comment]
> Even with
> 
> `bash <(curl -sSL https://spotx-official.github.io/run.sh) -c`
> 
> I had the same results.
> 
> I have the same results with uninstallify and reinstalling using the command below

When you ran uninstallify then SXB afterwards.. did you have to sign back into Spotify after it launched? Just want to confirm things were properly deleted since you didn't share the uninstallify output. 

if you re-run SXB with the `-e` flag to disable all experimental settings, does anything change?
```
bash <(curl -sSL https://spotx-official.github.io/run.sh) -ef
```

this can help narrow down if there's some experimental setting being used that your account doesn't like for some reason. if using `-e` doesn't change anything then we at least know it's not an experimental setting... but I still have no ideas what could be happening at the moment and am still unable to replicate this behavior. Both upgrading from the previous 1.2.47.x build.. and fully uninstalling everything and running the exact commands you used result in a fully working Spotify on my Arch, Debian and Ubuntu installs.

**Comment by fegauthier** [comment]
The issue seems to be about my network. If I enable a VPN, Spotify is working. If not, it doesnt work. Maybe my WAN IP was banned? I will take a look at my network.

**Comment by jetfir3** [comment]
> Spotify asked me to relogin yes. I tried with -ef but the result is the same. It's really weird.

> The issue seems to be about my network. If I enable a VPN, Spotify is working. If not, it doesnt work. Maybe my WAN IP was banned? I will take a look at my network.

Hmm.. it's even more strange because things seem to work with Spotify is NOT patched (right?) but breaks when patched with SXB. But... now it's fixed when VPN is used?! 

I don't doubt this feedback.. Spotify can do some weird things.. but I can't recall ever coming across this exact error from any users. 

If Spotify was broken, both with and without SXB, then it would make a little more sense. 
But with the following requiring a VPN to work:
```
bash <(curl -sSL https://spotx-official.github.io/run.sh)
```
and then the following allows it to work WITHOUT a VPN:
```
bash <(curl -sSL https://spotx-official.github.io/run.sh) --uninstall
```
... I'm quite confused. 

Please keep me posted of any findings and , if possible, be as detailed as you can.. and maybe this can get sorted. 
I doubt your IP was BANNED by Spotify.. but there could be some sort of CDN issue or something. 

- is the VPN server you're testing with also in Canada.. or another region?
- is your Spotify.com account set to Canada? https://www.spotify.com/account/profile/
- you have no issues accessing Spotify.com and signing in (with VPN OFF)?
- you have no issues accessing/using open.spotify.com web player (with VPN OFF)?

**Comment by amd64fox** [comment]
Perhaps the problem is related to this exp` enableBrotliCompressionViaCDN`

**Comment by fegauthier** [comment]
> > Spotify asked me to relogin yes. I tried with -ef but the result is the same. It's really weird.
> 
> > The issue seems to be about my network. If I enable a VPN, Spotify is working. If not, it doesnt work. Maybe my WAN IP was banned? I will take a look at my network.
> 
> Hmm.. it's even more strange because things seem to work with Spotify is NOT patched (right?) but breaks when patched with SXB. But... now it's fixed when VPN is used?!
> 
> I don't doubt this feedback.. Spotify can do some weird things.. but I can't recall ever coming across this exact error from any users.
> 
> If Spotify was broken, both with and without SXB, then it would make a little more sense. But with the following requiring a VPN to work:
> 
> ```
> bash <(curl -sSL https://spotx-official.github.io/run.sh)
> ```
> 
> and then the following allows it to work WITHOUT a VPN:
> 
> ```
> bash <(curl -sSL https://spotx-official.github.io/run.sh) --uninstall
> ```
> 
> ... I'm quite confused.
> 
> Please keep me posted of any findings and , if possible, be as detailed as you can.. and maybe this can get sorted. I doubt your IP was BANNED by Spotify.. but there could be some sort of CDN issue or something.
> 
> * is the VPN server you're testing with also in Canada.. or another region?
> * is your Spotify.com account set to Canada? https://www.spotify.com/account/profile/
> * you have no issues accessing Spotify.com and signing in (with VPN OFF)?
> * you have no issues accessing/using open.spotify.com web player (with VPN OFF)?

I'm pretty sure that the problem comes from my PC because on my laptop, it's working. I played with routing table this weekend. I probably messed with something. My laptop is using Zorin OS too and it's working. When I activate VPN on my PC. It plays with routing table. That's why it's working I guess.

**Comment by jetfir3** [comment]
> Perhaps the problem is related to this exp` enableBrotliCompressionViaCDN`
> 
> or with any other, sometimes they use convoluted descriptions
> 
> I would just compare the exp, what is enabled/disabled in the native client and compare it with what is in SpotX

As always, you've brought up some good ideas. Comparing remote config between stock and patched will be easiest route (assuming it's a remote config issue and not another patch made by SXB).


> I'm pretty sure that the problem comes from my PC because on my laptop, it's working. I played with routing table this weekend. I probably messed with something. My laptop is using Zorin OS too and it's working. When I activate VPN on my PC. It plays with routing table. That's why it's working I guess.

- is your laptop also running Linux
- you previous mentioned that things WERE working with STOCK but WEREN"T working after SXB patched Spotify. Are you 1000% sure that is the case? As that variable changes what may be needed to sort this issue out.

If you turn OFF the VPN, exit Spotify and then run the SXB uninstall command, the home screen loads properly?
```
bash <(curl -sSL https://spotx-official.github.io/run.sh) --uninstall
```
If it DOES load properly, we need to debug your Spotify remote config values.. between stock/unpatched and after patching. I'll have to give you steps to gather these values on stock and patched. 

If it DOESN'T load properly, then it would seem the issue is as you last described.. some network settings issue on your desktop.

**Comment by fegauthier** [comment]
> > Perhaps the problem is related to this exp` enableBrotliCompressionViaCDN`
> > or with any other, sometimes they use convoluted descriptions
> > I would just compare the exp, what is enabled/disabled in the native client and compare it with what is in SpotX
> 
> As always, you've brought up some good ideas. Comparing remote config and experimental features between stock and patched will be easiest route (assuming it's config/exp issue and not another patch made by SXB).
> 
> > I'm pretty sure that the problem comes from my PC because on my laptop, it's working. I played with routing table this weekend. I probably messed with something. My laptop is using Zorin OS too and it's working. When I activate VPN on my PC. It plays with routing table. That's why it's working I guess.
> 
> You previously mentioned that things WERE working with STOCK but WEREN'T working after SXB patched Spotify. Are you 1000% sure that is the case? As that specific variable changes what may be needed to sort this issue out.
> 
> If you turn OFF the VPN, exit Spotify and then run the SXB uninstall command, the home screen loads properly next time you launch Spotify?
> 
> ```
> bash <(curl -sSL https://spotx-official.github.io/run.sh) --uninstall
> ```
> 
> and then running the following causes the home screen to NOT load again?
> 
> ```
> bash <(curl -sSL https://spotx-official.github.io/run.sh) 
> ```
> 
> If the `--uninstall` command DOES load the Spotify home screen properly, we need to debug your Spotify remote config values.. between stock/unpatched and after patching. I'll have to give you steps to gather these values on stock and patched.
> 
> If it DOESN'T load properly, then it would seem the issue is as you last described.. some network settings issue on your desktop.

Content is loading on STOCK

![image](https://github.com/user-attachments/assets/1fe41d11-df0d-4cae-96ff-fc18b9c249b8)

But unable to play anything as it detects me offline

With PATCH, it's constantly loading.

I reapplied the patch and it's still loading. If I activate VPN again, it's working. When I deactivate the VPN. It's not working anymore. So it's a routing problem... Thanks guys for your support!

**Comment by jetfir3** [comment]
Since this issue seems to be caused by user editing routing tables on their device this issue will be closed for now.

**Comment by jaum20** [comment]
I have the same problem! maybe this is releated to the countries? Cause I am facing similar ussues with a patched android version of spotify (only works with vpn)

**Comment by jetfir3** [comment]
> I have the same problem! maybe this is releated to the countries? Cause I am facing similar ussues with a patched android version of spotify (only works with vpn)

Technically this could be related to the user's country... but only due to how the network's in those countries operate with GitHub servers. The issue discussed in this thread is the user having an issue connecting to the GitHub servers to download this script and nothing related to Spotify at all. 

Your issue with the Spotify mod on Android is 100% unrelated to this. And if the Android mod was properly patched, it would be working with issues. Instead, the current mods only provide temporary "band-aids" to fix an old version of Spotify and make it appear as new (spoofing the client version).. therefore this will break every time a new update is pushed out by Spotify requiring a new mod APK with new version spoof. A proper mod would not need a version spoof and fully work without the missing playlists, random shuffling, sign in issues etc etc. 

To summarize, your issue is unrelated to the issue in this thread and unrelated to SpotX/SpotX-Bash -- there is no need for further discussion here.

**Comment by jaum20** [comment]
>Technically this could be related to the user's country... but only due to how the networks in those countries operate with GitHub servers. The issue discussed in this thread is the user having an issue connecting to the GitHub servers to download this script and nothing related to Spotify at all

I have the problem of spotify no loading, like in the screenshot shared by @fegauthier in the first post. How is this related to github and not with spotify servers?

**Comment by jetfir3** [comment]
> > Technically this could be related to the user's country... but only due to how the networks in those countries operate with GitHub servers. The issue discussed in this thread is the user having an issue connecting to the GitHub servers to download this script and nothing related to Spotify at all
> 
> I have the problem of spotify no loading, like in the screenshot shared by @fegauthier in the first post. How is this related to github and not with spotify servers?

Yes, my mistake.. I confused this Issue with another recent Issue that was unrelated. Apologies.

However, the issue in this thread.. as shown by all the comments.. was due to a local computer routing configuration issue and unrelated to the users country and external network -- they had another computer on same network and same account that worked without issue.  

If you have a new issue which is confirmed to only be happening only after SpotX-Bash is applied and you've already attempted all [troubleshooting steps here](https://github.com/SpotX-Official/SpotX-Bash/wiki/Troubleshooting), open up a new Issue with as much detail, screenshots of SpotX installation etc as possible.

**Comment by jaum20** [comment]
I have tried it with two computers in two different networks with the same results (one of them fresh install). O will do some tests and open a new issue if it still do not work. Thanks
